### PR TITLE
feat: algokit deploy command integration and support for beaker template

### PR DESCRIPTION
**Pull request by aorumbayev**
# Proposed changes

- Integration of algokit deploy support 
- The following PR -> https://github.com/algorandfoundation/algokit-cli/pull/295 HAS to be merged first

### CODE REVIEW

**Comment by aorumbayev** [comment]
@achidlow @daniel-makerx feel free to check again

i removed the mnemonic to be set on common deploy since we don't need to prompt users for mnemonics on localnet deployments so its only a case on testnet

File: template_content/.algokit.toml.jinja [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+{%- elif deployment_language == 'typescript' %}
+command = "npm run deploy:ci"
+{%- endif %}
+environment_secrets = []
+
+[deploy.localnet]
+environment_secrets = []
+
+[deploy.testnet]
+environment_secrets = [
```
Comment (daniel-makerx) [inline template_content/.algokit.toml.jinja:16]: I think this should be back under the parent `[deploy]` section as that is a better indication that these secrets are used in all environments *except* localnet. Having it just in testnet implies that these secrets are only for use in testnet

File: template_content/.env.localnet.template.jinja [hunk 2]
```diff
@@ -1,3 +1,4 @@
+# this file should contain environment variables specific to algokit localnet
 ALGOD_TOKEN={{ algod_token }}
```
Comment (daniel-makerx) [inline template_content/.env.localnet.template.jinja:2]: This should also have indexer variables?

Comment (daniel-makerx) [reply template_content/.env.localnet.template.jinja:2]: Ignore that it does 😅

**Comment by aorumbayev** [comment]
closing in favor of merging together with the template simplification pr

**Pull request closed without merging**